plugins/item_tasks/server/json_tasks.py

```diff
@@ -0,0 +1,245 @@
+            'field of the item.')
+
+    jobModel = self.model('job', 'jobs')
+    token = self.model('token').createToken(
+        days=3, scope='item_task.set_task_spec.%s' % item['_id'],
+        user=self.getCurrentUser())
+    job = jobModel.createJob(
+        title='Read docker task specs: %s' % image, type='item.item_task_json_description',
+        handler='worker_handler', user=self.getCurrentUser())
+
+    jobOptions = {
+        'itemTaskId': item['_id'],
+        'kwargs': {
+            'task': {
+                'mode': 'docker',
+                'docker_image': image,
+                'container_args': [],
+                'pull_image': pullImage,
+                'outputs': [{
+                    'id': '_stdout',
+                    'format': 'text'
+                }],
+            },
+            'outputs': {
+                '_stdout': {
+                    'mode': 'http',
+                    'method': 'PUT',
+                    'format': 'text',
+                    'url': '/'.join((utils.getWorkerApiUrl(), 'item', str(item['_id']),
+                                     'item_task_json_specs')),
+                    'headers': {'Girder-Token': token['_id']},
+                    'params': {
+                        'image': image,
+                        'taskName': taskName,
+                        'setName': setName,
+                        'setDescription': setDescription,
+                        'pullImage': pullImage
+                    }
+                }
+            },
+            'jobInfo': utils.jobInfoSpec(job),
+            'validate': False,
+            'auto_convert': False,
+            'cleanup': True
+        }
+    }
+    job.update(jobOptions)
+
+    job = jobModel.save(job)
+    jobModel.scheduleJob(job)
+    return job
+
+
+@access.token
+@boundHandler()
+@autoDescribeRoute(
+    Description('Set a task spec on an item from a JSON specification.')
+    .modelParam('id', model='item', force=True)
+    .jsonParam('json', 'The JSON specifications as a list or a single object.',
+               paramType='body')
+    .param('image', 'The docker image name.', required=True, strip=True)
+    .param('taskName', 'The task name.', required=True, strip=True)
+    .param('setName', 'Whether item name should be changed to the name of the task.',
+           dataType='boolean', required=False, default=True)
+    .param('setDescription', 'Whether the item description should be changed to the '
+           'description of the task.', dataType='boolean', required=False, default=True)
+    .param('pullImage', 'Whether the image should be pulled from Docker Hub. '
+           'Set to false to use local images only.',
+           dataType='boolean', required=False, default=True),
+    hide=True
+)
+def configureItemTaskFromJson(self, item, json, image, taskName, setName, setDescription, pullImage,
+                              params):
+    self.ensureTokenScopes('item_task.set_task_spec.%s' % item['_id'])
```

Comment:
Based on the documentation for `ensureTokenScopes` (https://github.com/girder/girder/blob/637ab0e919bc355cbcb63a827eaabb8073c94408/girder/api/rest.py#L1026-L1029) the current behavior is unexpected. Indeed, this method implementation already passed review and is on master.

What would the proper implementation of `requireScopes` look like. Maybe it should be an API layer utility instead of on the model?